java/tools/src/java/org/apache/orc/tools/json/JsonSchemaFinder.java

```diff
@@ -58,7 +58,7 @@
   private static final Pattern TIMESTAMP_PATTERN =
       Pattern.compile("^[\"]?([0-9]{4}[-/][0-9]{2}[-/][0-9]{2})[T ]" +
           "([0-9]{2}:[0-9]{2}:[0-9]{2})" +
-          "(([ ][-+]?[0-9]{2}([:][0-9]{2})?)|Z)?[\"]?$");
+          "(([ ]?[-+]?[0-9]{2}([:][0-9]{2})?)|Z)?[\"]?$");
```

Comment:
I'd propose that this should be "( [-+]?|[-+])" instead of "[ ]?[-+]?". It doesn't seem right to accept 
"2016-12-30T12:34:5608"